cpp/src/copying/scatter.cu

```diff
@@ -177,6 +179,88 @@ void scalar_scatter(const std::vector<gdf_scalar>& source,
 
 }
 
+inline bool validate_scatter_map(gdf_column const& scatter_map,
+                          cudf::table const& input_table) {
+  CUDF_EXPECTS(scatter_map.dtype == GDF_INT32,
+      "scatter_map is not GDF_INT32 column.");
+  CUDF_EXPECTS(not cudf::has_nulls(scatter_map),
+      "Scatter map cannot contain null elements.");
+  CUDF_EXPECTS(scatter_map.size == input_table.num_rows(),
+      "scatter_map length is not equal to number of rows in input table.");
+  if (scatter_map.size == 0 ||
+      input_table.num_columns() == 0 ||
+      input_table.num_rows() == 0)
+    return false;
+  return true;
+}
+
+template<typename InputIterator>
+void groups_to_tables(cudf::table const& input_table,
+                       gdf_index_type* scatter_array,
+                       InputIterator first,
+                       InputIterator last,
+                       std::vector<cudf::table>& output_tables) {
+  size_t num_groups = thrust::distance(first, last);
+  output_tables.reserve(num_groups);
+  for (auto iter=first; iter<last; iter++) {
+    typename std::iterator_traits<InputIterator>::value_type groupid = *iter; 
+    output_tables.push_back(
+        detail::copy_if(input_table,
+          [scatter_array, groupid] __device__ (gdf_index_type row)
+          { return groupid==scatter_array[row];
+          }));
+  }
+}
+
+void groups_to_tables(cudf::table const& input_table,
```

Comment:
While this is legal, it's pretty confusing to have two functions with the same name in the same namespace.